### PR TITLE
feat: Create SQLAlchemy URL from config take 2

### PR DESCRIPTION
**Pull request by BuzzCutNorman**
This is an attempt to satisfy the below feature that is mentioned in issue #4 

>  Host,Username,Port,Password,DBName, with SQL Alchemy having folks specify a url may be ok for now, it's the same thing. Seperate issue potentially

### CODE REVIEW

**Comment by BuzzCutNorman** [comment]
@visch I believe this matches up exactly with the previous PR.

File: target_postgres/connector.py [hunk 1]
```diff
@@ -30,6 +31,37 @@ def create_sqlalchemy_connection(self) -> sqlalchemy.engine.Connection:
         """
         return self.create_sqlalchemy_engine().connect()
 
+    def get_sqlalchemy_url(self, config: dict) -> str:
+        """Generates a SQLAlchemy URL for sqlbuzz.
+        Args:
+            config: The configuration for the connector.
+        """
+        if config['dialect'] == "postgresql" :
```
Comment (visch) [inline target_postgres/connector.py:39]: This implementation is pretty specific to `psycopg2` in a few places. I think ARRAY(JSON) only works with psycopg2. Not certain of that but I believe so. I think we should remove dialect as an option

File: target_postgres/connector.py [hunk 2]
```diff
@@ -30,6 +31,37 @@ def create_sqlalchemy_connection(self) -> sqlalchemy.engine.Connection:
...
+            url_drivername:str = config['dialect']
```
Comment (visch) [inline target_postgres/connector.py:40]: I like giving folks the option to switch but I think in like a full overridable sql alchemy url makes a ton of sense as there's proably N number of things you can do with sqlalchemy urls all of which are things we may not have planned for with our strict host/user config options. Maybe we add a `sqlalchemy` config option and the functionality will be to override the sqlalchemy url (ignoring host, user,etc)

File: target_postgres/connector.py [hunk 3]
```diff
@@ -30,6 +31,37 @@ def create_sqlalchemy_connection(self) -> sqlalchemy.engine.Connection:
...
+            exit(1)
+
+        if config['driver_type'] in ["psycopg2","pg8000","asyncpg","psycopg2cffi","pypostgresql","pygresql"]:
+            url_drivername += f"+{config['driver_type']}"
+        else:
+            self.logger.error("Invalid driver_type given")
+            exit(1)
+
+        self.logger.info(url_drivername)
+        config_url = URL.create(
```
Comment (visch) [inline target_postgres/connector.py:52]: I'm a little worried that we aren't flexible enough here. I think allowing a sqlalchemy url override solves my issue. There's probably options for authing over TLS and other auth options, and connection options available and we're being pretty strict about how we want you to connect to the db.

File: target_postgres/target.py [hunk 4]
```diff
@@ -12,11 +12,39 @@ class TargetPostgres(Target):
...
-            + "`postgresql://postgres:postgres@localhost:5432/postgres`",
+            description="The Dialect of SQLAlchamey"
+        ),
+        th.Property(
+            "driver_type",
+            th.StringType,
+            description="The Python Driver you will be using to connect to the SQL server"
+        ),
+        th.Property(
+            "host",
```
Comment (visch) [inline target_postgres/target.py:25]: required=True for this, user, password, and database I think and we'd be good. If they are not set we'd fail with a keyerror from the function above anyway

Comment (visch) [reply target_postgres/target.py:52]: With sqlalchemy url's also being in the list we can't just require these if the other isn't required.

**Comment by BuzzCutNorman** [comment]
@visch The changes look great.

**Pull request merged**